src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt

```diff
@@ -530,6 +527,54 @@ class CCv2Service(val project: Project, private val coroutineScope: CoroutineSco
         }
     }
 
+    fun fetchBuildWithCode(
+        subscription: CCv2Subscription,
+        buildCode: String,
+        onStartCallback: () -> Unit,
+        onCompleteCallback: (CCv2BuildDto) -> Unit
+    ) {
+        onStartCallback.invoke()
+
+        coroutineScope.launch {
+            withBackgroundProgress(project, "Fetching Build...$buildCode", true) {
+                val ccv2Token = getCCv2Token(subscription) ?: return@withBackgroundProgress
+                try {
+                    val build = CCv2Api.getInstance().fetchBuildForCode(ccv2Token, subscription, buildCode)
+                    onCompleteCallback.invoke(build)
+                } catch (e: SocketTimeoutException) {
+                    notifyOnTimeout(subscription)
+                } catch (e: RuntimeException) {
+                    notifyOnException(subscription, e)
+                }
+            }
+        }
+    }
+
+    fun fetchDeploymentsForBuild(
+        subscription: CCv2Subscription,
+        buildCode: String,
+        onStartCallback: () -> Unit,
+        onCompleteCallback: (List<CCv2DeploymentDto>) -> Unit
+    ) {
+        onStartCallback.invoke()
+        var deployments: List<CCv2DeploymentDto>
+        coroutineScope.launch {
+            withBackgroundProgress(project, "Fetching Deployment for build...", true) {
```

Comment:
add buildCode, like `Fetching Deployment for build - $buildCode...`

Reply:
Resolved.